### PR TITLE
[GH-158] Add 'pretty' flag to support formatting of downloaded AVRO and JSON schemas

### PR DESCRIPTION
**Pull request by filpano**
I haven't worked much with Kotlin so I tried to mostly keep the code style consistent with the rest. If there's any style choices you're not quite satisfied with, feel free to suggest changes and/or modify.

Protobuf schema downloads include newlines and hence are, in fact, already formatted. Regardless, so as to not be a 'soft' breaking change, I've made the `pretty` option `false` by default. Setting a default of `true` would be more consistent with the protobuf handling though. 

Re: tests: I added both a unit test and an integration test but I'm not quite happy with the integration test at the moment since I didn't want to spend too much time on it. Feedback are welcome.

One last thing: I didn't have a chance to test *registering* a formatted avro/json schema. I know that the Schema Registry API does _not_ support it and expects an escaped JSON string - I'm not sure if the programmatic client calls support it. Still need a test case for that...

### CODE REVIEW

**Comment by filpano** [comment]
I also just noticed that I forgot to update the `README` - I can do it tomorrow.

File: src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt [hunk 1]
```diff
@@ -93,6 +100,69 @@ class DownloadTaskIT : KafkaTestContainersUtils() {
...
+
+        Assertions.assertThat(File(folderRule.root, "src/main/$type/test_v1")).exists()
+        val resultFile1 = File(folderRule.root, "src/main/$type/test_v1/$schemaFile")
+        Assertions.assertThat(resultFile1).exists()
+        Assertions.assertThat(resultFile1.readText()).doesNotContain("description")
+
+        Assertions.assertThat(File(folderRule.root, "src/main/$type/test_v2")).exists()
+        val resultFile2 = File(folderRule.root, "src/main/$type/test_v2/$schemaFile")
+        Assertions.assertThat(resultFile2).exists()
+        Assertions.assertThat(resultFile2.readText()).contains("description")
```
Comment (ImFlog) [inline src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt:163]: Meybe to simplify this test and only check the formatting you could remove the "new schema" and only register one schema and verify that it's retrieved correctly ?

File: src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt [hunk 2]
```diff
@@ -93,6 +100,69 @@ class DownloadTaskIT : KafkaTestContainersUtils() {
...
     }
 
+    @ParameterizedTest(name = "{0}")
+    @ArgumentsSource(SchemaArgumentProvider::class)
+    fun `Should format supported schema types when pretty is specified`(type: String, oldSchema: ParsedSchema,
+                                                                        newSchema: ParsedSchema) {
+        // Given
+        val subjectName = "parameterized-$type"
+
+        client.register(subjectName, oldSchema)
```
Comment (ImFlog) [inline src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt:110]: To harden the test, can you remove all the new line when registering the schema ?
I don't know if the mock registry client do any formatting on the passed schemas.

**Review by ImFlog — COMMENTED** [review]
Great PR !
Nothing much to say apart from a small simplification of the integration test. Tell me if this makes sense for you.
Once the README is updated I think we will be able to merge this quickly.

File: src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt (hunk 2 above)
Comment (filpano) [reply src/integration/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskIT.kt:110]: Good idea, will do.

**Comment by ImFlog** [comment]
LGTM except for the failing the test.
I will try to take time to fix the test ASAP.

**Comment by filpano** [comment]
I've managed to narrow the issue down to commit 04e52e3c - though I'm not sure how since I didn't touch the failing test in that commit. I can take a quick look.

**Comment by filpano** [comment]
The above assumption was correct. I'm not 100% sure why the `client.reset()` didn't work if it was registered under the same name in that case, but I did verify that the `@Teardown` method is called after every iteration of the parametrized test. ¯\_(ツ)_/¯

**Comment by ImFlog** [comment]
LGTM !
Tests are green now, next time I get the chance to commit something I will try to take a deeper look at the reset issue in tests.
I will try to release a new version, maybe today if I can find a bit of time.

Thank you again for the contribution 🙇

**Pull request merged**